pkg/pool-stable-phantom/contracts/StablePoolRates.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IRateProvider.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/InputHelpers.sol";
+import "@balancer-labs/v2-pool-utils/contracts/rates/PriceRateCache.sol";
+
+abstract contract StablePoolRates {
```

Comment:
I think it's fine to have this inherit from storage to be honest. That'll let us add many other functions, such as `getTokenRateCache`, `getTokenRate`, `_getAdjustedBalances`, `setTokenRateCacheDuration`, etc. As is, this module is not too useful for testing responsability removal purposes.

Down the line, if say we had multiple stable pools, we could then abstract away storage functions into virtual ones (like we do in BaseWeightedPool), but we don't need to worry about that now.

Reply:
Ok; can try it that way. (Getting very close to the original "aggressive" PR, though the helper trick is an improvement.)